src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,170 @@
+import SearchResults from '../components/SearchResults/SearchResults';
+import { placeholderText, currentItemIDInitial } from './SearchPageVariables';
+import Pagination from '../components/Pagination/Pagination';
+import ItemsSelect from '../components/ItemsPerPage/ItemsSelect';
+import DetailsSection from '../components/DetailsSection/DetailsSection';
+
+export function SearchPage() {
+  const [query, setQuery] = useState('');
+  const [searchResults, setsearchResults] = useState<SearchResultType[]>([]);
+  const [isSearchLoading, setIsSearchLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [totalPages, setTotalPages] = useState(0);
+  const [limit, setLimit] = useState(10);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [openDetails, setOpenDetails] = useState(false);
+  const [currentItemID, setCurrentItemID] = useState('');
+  const [currentItem, setCurrentItem] =
+    useState<SearchResultType>(currentItemIDInitial);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const savedQuery = localStorage.getItem('searchQuery');
+    if (savedQuery) {
+      getSearchResults(savedQuery, limit, currentPage);
+    } else getSearchResults(query, limit, currentPage);
+  }, []);
+
+  useEffect(() => {
+    getSearchResults(query, limit, currentPage);
+  }, [currentPage, limit]);
+
+  useEffect(() => {
+    setCharacterDetails();
+  }, [currentItemID]);
+
+  useEffect(() => {}, [currentItem]);
+
+  useEffect(() => {
+    if (hasError) {
+      throw new Error('oh no! It is an Error!');
+    }
+  }, [hasError]);
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setQuery(event.target.value);
+  }
+
+  function changePage(page: number) {
+    setCurrentPage(page);
+    navigate(`/results/${page}`);
+  }
+
+  function changeItemsLimit(event: React.ChangeEvent<HTMLSelectElement>) {
+    setLimit(Number(event.target.value));
+  }
+  function handleClick(e: React.MouseEvent<HTMLDivElement>) {
+    if (openDetails) {
+      const className = (e.target as Element).className;
+      if (className === 'background') {
+        setOpenDetails(false);
+      }
+    }
+  }
+  async function getSearchResults(
+    searchStr: string,
+    limit: number,
+    currentPage: number
+  ) {
+    setIsSearchLoading(true);
+    try {
+      const responseJSON = await APISearch(searchStr, limit, currentPage);
+      const resultsJSON = responseJSON.docs;
+      const totalPagesData = responseJSON.pages;
+      setTotalPages(totalPagesData);
+
+      const results = resultsJSON.map((item: SearchResultType) => ({
+        name: item.name,
+        height: item.height,
+        race: item.race,
+        birth: item.birth,
+        spouse: item.spouse,
+        _id: item._id,
+      }));
+      setsearchResults(results);
+      setIsSearchLoading(false);
+    } catch (error) {
+      console.log(error);
+      setIsSearchLoading(false);
+      setHasError(true);
+    }
```

Comment:
You do the same on lines 91 and 94
You can use `finally` for this
exmp
` try {  ..... }`
`catch (e) { ..... }`
`finally {setIsSearchLoading(false);}`